### ISSUE TITLE
[x86] [JITMinOpts] Failing test smallFrame

### ISSUE THREAD

**Issue by LLITCHEV** [dotnet/coreclr#4548]
The test d:\repos\coreclr\tests\src\TestWrappers_x86_Checked\JIT\JIT.XUnitWrapper.cs(281625): error : JIT._Methodical_tailcall_v4_smallFrame_smallFrame_._Methodical_tailcall_v4_smallFrame_smallFrame_cmd [FAIL] [d:\repos\coreclr\tests\runtest.proj] fails when run in JITMinOpts on CoreCLR for x86.

This issue could be a duplicate of Issue #2353 - https://github.com/dotnet/coreclr/issues/2353

**Comment by sejongoh** [comment]
This passes on Desktop.

```
set CORE_ROOT=C:\Windows\Microsoft.NET\Framework\v4.0.30319
; open smallFrame.il and replace "[System.Console]" with "[mscorlib]".
ilasm smallFrame.il
.\smallFrame.exe
```

**Comment by sejongoh** [comment]
It looks like I didn't set COMPlus_JITMintOpt in the previous run. 

This test fails on Desktop, ProjectK and CoreCLR with the same error message.

```
C:\Workspaces\ws1\coreclr\tests\src\JIT\Methodical\tailcall_v4>set COM
CommandPromptType=Cross
CommonProgramFiles=C:\Program Files\Common Files
CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files
CommonProgramW6432=C:\Program Files\Common Files
COMPlus_JITMinOpts=1
COMPUTERNAME=SEJOOH-PC
ComSpec=C:\windows\system32\cmd.exe

C:\Workspaces\ws1\coreclr\tests\src\JIT\Methodical\tailcall_v4>.\s.exe

Unhandled Exception: System.InvalidProgramException: Common Language Runtime detected an invalid program.
   at RetBufferBug.TailCaller()
   at RetBufferBug.Main()
```

**Comment by sejongoh** [comment]
@RussKeldorph To which milestone can we move this?

**Comment by RussKeldorph** [comment]
@sejongoh This was excluded under minopts in the old harness, so we should be able to resolve the bug  by marking it "optimization sensitive."  I'll take the bug since I'm already fixing another similar test.

**Comment by RussKeldorph** [comment]
Disabled via #5057.  I'm not certain this isn't a bug in JIT32, so I'm moving it to Future instead of closing.

**Comment by RussKeldorph** [comment]
Should not have been closed, just moved to Future.

**Comment by sivarv** [comment]
The failing test has the following tail call pattern

Tail.call
Pop
Ret

This tail call pattern is supported only on amd64 for Legacy Jit64 compat reasons and not supported on x86.  For that reason I would expect this test to fail with InvalidProgramException on x86/Arm32.

**Comment by pgavlin** [comment]
Closed by #11110.
